File: mockez-back/src/main/java/com/mockez/service/CategoryService.java
package com.mockez.service;

import com.mockez.domain.model.entity.Category;

import java.util.List;
import java.util.UUID;

public interface CategoryService {

    /**
     * CATEGORY.01 GET /category
     * @return categories
     */
    List<Category> getCategories();

    /**
     * CATEGORY.02 GET /category/{id}
     * @param id of category
     * @return category
     */
    Category getCategoryHasGenerateTypes(UUID id);

    /**
     * CATEGORY.03 GET /category/{id}/generate-types/count
     * @param id of category
     * @return number of generate-types in category
     */
    Integer getGenerateTypesCount(UUID id);
}

File: mockez-front/src/app/core/service/category.service.ts
import { Injectable } from '@angular/core';
import { Category } from '@core/model/category';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { environment } from '@environment/environment';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  constructor(
    private httpClient: HttpClient
  ) {
  }

  getCategories(): Observable<Category[]> {
    return this.httpClient.get<Category[]>(`${environment.apiUrl}/category`);
  }

  getCategory(id: string): Observable<Category> {
    return this.httpClient.get<Category>(`${environment.apiUrl}/category/${id}`);
  }

  getGenerateTypesCount(id: string): Observable<number> {
    return this.httpClient.get<number>(`${environment.apiUrl}/category/${id}/generate-types/count`);
  }
}
